packages/dev/core/src/scene.ts

```diff
@@ -1808,6 +1816,12 @@ export class Scene extends AbstractScene implements IAnimatable, IClipPlanesHold
         return this._cachedEffect !== effect || this._cachedMaterial !== material || this._cachedVisibility !== visibility;
     }
 
+    /**
+     * Gets the engine associated with the scene
+     * @returns an Engine
+     */
+    public getEngine<T extends Engine>(): T;

```

Comment:
Let s not add this one. The type changes at runtime depending on the running environment.